src/Compilers/VisualBasic/Portable/Symbols/SymbolExtensions.vb

```diff
@@ -380,14 +380,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
         <Extension>
         Friend Function GetDeclaringSyntaxNode(Of T As VisualBasicSyntaxNode)(this As Symbol) As T
 
-            For Each node In this.DeclaringSyntaxReferences.Select(Function(d) d.GetSyntax())
-                Dim node_T = TryCast(node, T)
+            For Each reference In this.DeclaringSyntaxReferences
+                Dim node_T = TryCast(reference.GetSyntax(), T)
                 If node_T IsNot Nothing Then
                     Return node_T
                 End If
             Next
 
-            Debug.Assert(this.IsImplicitlyDeclared)
+            Debug.Assert(this.IsImplicitlyDeclared OrElse (TypeOf this Is SourceMemberContainerTypeSymbol And
```

Comment:
>TypeOf this Is SourceMemberContainerTypeSymbol [](start = 59, length = 46)

The type cast is performed twice. Consider using conditional access off of ```TryCast(this, SourceMemberContainerTypeSymbol)``` instead.